schema-derivatives-perpfutures.graphql

```diff
@@ -0,0 +1,1425 @@
+  withdraws: [Withdraw!]! @derivedFrom(field: "account")
+
+  " All Position events occurred in this account "
+  positions: [Position!]! @derivedFrom(field: "account")
+
+}
+
+
+# Helper entity for calculating daily/hourly active users
+type ActiveAccount @entity{
+  " { daily/hourly }-{ Address of the account }-{ Days/hours since Unix epoch } "
+  id: Bytes!
+
+   hourlyActiveUsers: Int!
+  dailyActiveUsers: Int!
+}
+
+# A position is defined as who has what position in the protocol and distribution of longs and shorts at a positional level
+type Position @entity {
+  " { Account address }-{ Market address }-{ Position Side }-{ Optional: Interest Rate Type}-{ Counter } "
+  id: ID!
+
+  " Account that owns this position "
+  account: Account!
+
+  " The market in which this position was opened "
+  market: Market!
+
+  " The asset in which this position was opened with "
+  asset: Token!
+
+  " The hash of the transaction that opened this position "
+  hashOpened: Bytes!
+
+  " The hash of the transaction that closed this position "
+  hashClosed: Bytes
+
+  " Block number of when the position was opened "
+  blockNumberOpened: BigInt!
+
+  " Timestamp when the position was opened "
+  timestampOpened: BigInt!
+
+  " Block number of when the position was closed (0 if still open) "
+  blockNumberClosed: BigInt
+
+  " Timestamp when the position was closed (0 if still open) "
+  timestampClosed: BigInt
+
+  " Side of the position (either LP or trader) "
+  side: PositionSide!
+
+  " Funding rate (premium) for position when position open for trader in USD"
+  fundingrateOpen: BigInt
+
+" Funding rate (premium) for position when position closed for trader in USD"
+  fundingrateClosed: BigInt
+
+  " Token balance in this position, in native amounts "
+  balance: BigInt!
+
+  " Number of deposits related to this position "
+  depositCount: Int!
+
+  " All deposit events of this position "
+  deposits: [Deposit!]! @derivedFrom(field: "position")
+
+  " Number of withdrawals related to this position "
+  withdrawCount: Int!
+
+  " All withdraw events of this position "
+  withdraws: [Withdraw!]! @derivedFrom(field: "position")
+
+  " Number of longs related to this position  "
+  longCount: Int!
+
+  " Long event related to this position "
+  long: [Long!]! @derivedFrom(field: "position")
+
+  " Number of shorts related to this position  "
+  shortCount: Int!
+
+  " Short event related to this position "
+  short: [Long!]! @derivedFrom(field: "position")
+
+  " Number of liquidations related to this position (incremented when this position is liquidated) "
+  liquidationCount: Int!
+
+  " Liquidation event related to this position (if exists) "
+  liquidations: [Liquidate!]! @derivedFrom(field: "position")
+
+  " Position daily snapshots for open positions "
+  snapshots: [PositionSnapshot!]! @derivedFrom(field: "position")
+}
+
+type positionDailySnapshot @entity(immutable: true) {
```

Comment:
Position snapshots we usually take them on change instead of periodically. To avoid the number of snapshots growing immensely and because it will be very hard to implement